mars/services/scheduling/worker/execution.py

```diff
@@ -48,16 +50,19 @@ class SubtaskExecutionInfo:
     supervisor_address: str
     result: SubtaskResult = field(default_factory=SubtaskResult)
     cancelling: bool = False
+    max_runs: int = 1
+    num_runs: int = 0
     slot_id: Optional[int] = None
     kill_timeout: Optional[int] = None
 
 
 class SubtaskExecutionActor(mo.Actor):
     _subtask_info: Dict[str, SubtaskExecutionInfo]
 
-    def __init__(self):
+    def __init__(self, default_config: Dict):
```

Comment:
We recommend adding arguments for actor constructors directly. Service config can be parsed and loaded to actors via start() methods.

Reply:
The `default_config` is similar with the `WorkerQuotaManagerActor`: https://github.com/mars-project/mars/blob/master/mars/services/scheduling/worker/service.py#L47

Reply:
> For some operands (esp. operands with user code) we have `retry_when_fail` argument which defines the behavior of operand after fail. Subtask retrial need to take that into consideration.

Thanks. I will take account into  the `retry_when_fail` argument.

Reply:
@wjsi Is the `retry_when_fail` implemented? I can't find the retry logic.

Reply:
> The `default_config` is similar with the `WorkerQuotaManagerActor`: https://github.com/mars-project/mars/blob/master/mars/services/scheduling/worker/service.py#L47

`default_config` for `WorkerQuotaManagerActor` is designed to create other actors and can be overwritten. For `SubtaskExecutionActor`, there is no defaults. Hence please use arguments directly.

Reply:
> > The `default_config` is similar with the `WorkerQuotaManagerActor`: https://github.com/mars-project/mars/blob/master/mars/services/scheduling/worker/service.py#L47
> 
> `default_config` for `WorkerQuotaManagerActor` is designed to create other actors and can be overwritten. For `SubtaskExecutionActor`, there is no defaults. Hence please use arguments directly.

Got it. I will change the `default_config` to the arguments.